src/main/java/org/opentripplanner/model/plan/ItinerariesCalculateLegTotals.java

```diff
@@ -39,6 +41,9 @@ else if(leg.isOnStreetNonTransit()) {
             if(!leg.isWalkingLeg()) {
                 walkOnly = false;
             }
+            if(leg.isTransitLeg()) {
+              this.streetOnly = false;
+            }
```

Comment:
Check the mode `LegSwitch`, is it street or transit, none or both?

Reply:
LEG_SWITCH is new explicit ignored, witch is correct if the LEG_SWITCH is ever reintroduced.